src/wallet/sigmaspendbuilder.cpp

```diff
@@ -105,6 +105,10 @@ static std::unique_ptr<SigmaSpendSigner> CreateSigner(const CSigmaEntry& coin)
 
 SigmaSpendBuilder::SigmaSpendBuilder(CWallet& wallet, const CCoinControl *coinControl) : TxBuilder(wallet)
 {
+    if (!zwalletMain) {
+        throw std::logic_error("Sigma feature required HD wallet");
```

Comment:
also I think better to create guard function instead. because we use this several time.